src/controller/CHIPDeviceController.cpp

```diff
@@ -2777,6 +2794,187 @@ CHIP_ERROR DeviceCommissioner::ParseICDInfo(ReadCommissioningInfo & info)
+        using namespace chip::app::Clusters::JointFabricAdministrator::Attributes;
+        AdministratorFabricIndex::TypeInfo::DecodableType administratorFabricIndex;
+
+        VerifyOrReturnError(path.mAttributeId == AdministratorFabricIndex::Id, CHIP_NO_ERROR);
+        ReturnErrorOnFailure(this->mAttributeCache->Get<AdministratorFabricIndex::TypeInfo>(path, administratorFabricIndex));
+
+        if (!administratorFabricIndex.IsNull())
+        {
+            ChipLogProgress(Controller, "JCM: AdministratorFabricIndex: %d", administratorFabricIndex.Value());
+            info.JFAdministratorFabricIndex = administratorFabricIndex.Value();
+            info.JFAdminEndpointId          = path.mEndpointId;
+        }
+        else
+        {
+            ChipLogError(Controller, "JCM: AdministratorFabricIndex attribute@JF Administrator Cluster not found!");
+            return CHIP_ERROR_NOT_FOUND;
+        }
+        return CHIP_NO_ERROR;
+    });
+
+    if (err != CHIP_NO_ERROR)
+    {
+        return err;
+    }
+
+    err = mAttributeCache->ForEachAttribute(
+        OperationalCredentials::Id, [this, &info, &rootKeySpan](const ConcreteAttributePath & path) {
+            using namespace chip::app::Clusters::OperationalCredentials::Attributes;
+
+            switch (path.mAttributeId)
+            {
+            case Fabrics::Id: {
+                Fabrics::TypeInfo::DecodableType fabrics;
+                ReturnErrorOnFailure(this->mAttributeCache->Get<Fabrics::TypeInfo>(path, fabrics));
+
+                auto iter = fabrics.begin();
+                while (iter.Next())
+                {
+                    auto & fabricDescriptor = iter.GetValue();
+                    if (fabricDescriptor.fabricID == info.JFAdministratorFabricIndex)
+                    {
+                        if (fabricDescriptor.rootPublicKey.size() != Crypto::kP256_PublicKey_Length)
+                        {
+                            ChipLogError(Controller,
+                                         "JCM: DeviceCommissioner::ParseJFAdministratorInfo - fabric root key size mismatch");
+                            return CHIP_ERROR_KEY_NOT_FOUND;
+                        }
+                        rootKeySpan                      = fabricDescriptor.rootPublicKey;
+                        info.JFAdminFabricTable.vendorId = fabricDescriptor.vendorID;
+                        info.JFAdminFabricTable.fabricId = fabricDescriptor.fabricID;
+
+                        if (fabricDescriptor.VIDVerificationStatement.HasValue())
+                        {
+                            ChipLogError(Controller, "JCM: Per-home RCAC are not supported by JF for now!");
+                            return CHIP_ERROR_CANCELLED;
+                        }
+                        ChipLogProgress(Controller, "JCM: Successfully parsed the Administrator Fabric Table");
+                        break;
+                    }
+                }
+                return CHIP_NO_ERROR;
```

Comment:
This is never checking whether we got a value but it's not well-formed TLV (in which case Next() will return false at some point and the status of the iterator will be a failure).

Reply:
Indeed, fixed this not by checking the iterator status but by making sure that the fields gets populated. In the end, if the fields don't get populated, it's not of interest what was the root issue of that.

Reply:
How do you want to handle the case when the fields get populated but the TLV is still invalid _after_ that point?

Reply:
I guess that works because you stop iterating at that point, ok.